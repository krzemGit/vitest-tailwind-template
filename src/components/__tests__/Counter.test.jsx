import { beforeEach, describe, expect, test } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import Counter from "../Counter";

describe("Counter", () => {
  let snapshot;

  beforeEach(() => {
    snapshot = render(<Counter title="Testing" />);
  });

  test("should match the snapshot", () => {
    expect(snapshot.container).toMatchSnapshot();
  });

  test("should show title all the time", () => {
    expect(screen.getByText(/Testing/i)).toBeDefined();
  });

  test("should increase the number", () => {
    const zeroNumber = screen.getByText("0");
    expect(zeroNumber.innerHTML).toEqual("0");

    const addBtn = screen.getByText(/count/i);
    fireEvent.click(addBtn);

    expect(zeroNumber.innerHTML).toEqual("1");
  });
});
