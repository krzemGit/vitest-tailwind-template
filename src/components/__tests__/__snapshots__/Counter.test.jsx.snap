// Vitest Snapshot v1

exports[`Counter > should match the snapshot 1`] = `
<div>
  <div
    class="flex h-screen"
  >
    <h1>
      Testing
    </h1>
    <div
      class="m-auto"
    >
      <div
        class="text-6xl text-red-600"
      >
        0
      </div>
      <button
        class="px-6 py-2 rounded bg-green-800 hover:bg-green-600 text-white"
        type="button"
      >
        count+
      </button>
    </div>
  </div>
</div>
`;
